src/layers/ConstructionLayer/ConstructionLayer.js

```diff
@@ -0,0 +1,202 @@
+import OLVectorLayer from 'ol/layer/Vector';
+import OLVectorSource from 'ol/source/Vector';
+import ClusterSource from 'ol/source/Cluster';
+import GeoJSON from 'ol/format/GeoJSON';
+import Style from 'ol/style/Style';
+import Icon from 'ol/style/Icon';
+import Text from 'ol/style/Text';
+import Fill from 'ol/style/Fill';
+import Circle from 'ol/style/Circle';
+import { unByKey } from 'ol/Observable';
+
+/**
+ * Layer for construction
+ * Extends {@link https://react-spatial.geops.de/docjs.html#layer geops-spatial/Layer}
+ * @class
+ * @param {Object} [options] Layer options.
+ */
+class ConstructionLayer extends VectorLayer {
+  constructor(options = {}) {
+    const olLayer = new OLVectorLayer({
+      style: (f, r) => this.style(f, r),
+      source: new OLVectorSource({
+        format: new GeoJSON(),
+        loader: () => {
+          fetch(
+            `${this.geoServerUrl}?` +
+              'service=WFS&version=1.0.0&request=GetFeature&' +
+              'typeName=trafimage:bahnausbauten&' +
+              'srsName=EPSG:3857&' +
+              'outputFormat=application/json',
+          )
+            .then(data => data.json())
+            .then(data => {
+              const format = new GeoJSON();
+              const features = format.readFeatures(data);
+              this.getSource().clear();
+              this.getSource().addFeatures(features);
+            });
+        },
+      }),
+      minResolution: options.minResolution,
+      maxResolution: options.maxResolution,
+    });
+
+    super({
+      ...options,
+      olLayer,
+    });
+
+    this.cluster = this.get('cluster');
+
+    this.styleCache = {};
+    this.visibilityKeys = [];
+
+    this.onChangeVisible = this.onChangeVisible.bind(this);
+    this.geometryFunction = this.geometryFunction.bind(this);
+    this.getSource = this.getSource.bind(this);
+
+    this.setVisible(this.visible);
+
+    if (this.cluster) {
+      this.olLayer.setSource(
+        new ClusterSource({
+          distance: 90,
+          source: this.olLayer.getSource(),
+          geometryFunction: this.geometryFunction,
+        }),
+      );
+    }
+  }
+
+  init(map) {
+    super.init(map);
+
+    this.visibilityKeys.push(
+      this.children.map(childLayer => {
+        return childLayer.on('change:visible', this.onChangeVisible);
+      }),
+    );
+  }
+
+  terminate() {
+    super.terminate();
+
+    if (this.visibilityKeys.length) {
+      unByKey(this.visibilityKeys);
+    }
+  }
+
+  getFeatureInfoAtCoordinate(coordinate) {
+    return super.getFeatureInfoAtCoordinate(coordinate).then(data => {
+      const features = data.features
+        .map(f => (f.get('features') ? f.get('features') : f))
+        .flat();
+
+      return Promise.resolve({
+        features,
+        layer: data.layer,
+        coordinate: data.features.length
+          ? data.features[0].getGeometry().getCoordinates()
+          : data.coordinate,
+      });
```

Comment:
In the Popup component, this coordinate isn't used as all features are Points (see [line](https://github.com/geops/trafimage-maps/blob/master/src/components/Popup/Popup.js#L28)). For clusters, this results in strange positions of the popup (if we use this approach with `getFeatureInfoAtCoordinate()`). @oterral Would it be possible always use a given coordinate in a feature info?

Reply:
Can't you get the cluster info ?

Reply:
@oterral I would like, but it is always overridden by the geometry coordinate of the first feature, see [line](https://github.com/geops/trafimage-maps/blob/master/src/components/Popup/Popup.js#L28)

Reply:
I guess you can change that in Popup and take the centroid of all features.